### PR TITLE
Fix GC tests failing with default read mode

### PR DESCRIPTION
**Pull request by agarwal-navin**
As part of [#3904](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3904), container is loaded in "read" mode be default. Some GC tests were failing with this change so a workaround was added to load the container in "write" mode for tests tests.

This PR removes that workaround and makes the tests work with "read" mode containers. There are 2 ways these tests are fixed:
1. The test in `gcInactiveNodes.spec.ts` requires the container to be in "write" mode so that the GC data in base snapshot is loaded and marks unreferenced nodes as inactive. So, updated the test to send an op and wait for "write" connection.
2. The other tests were not testing the inactive / sweep ready state but rather tombstone / deleted state. For these, the container does not need to load in "write" mode. So, for these tests removed the expected inactive / sweep ready errors since they won't happen in loaded containers anymore.

[AB#3982](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/3982)

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts [hunk 1]
```diff
@@ -313,16 +314,21 @@ describeNoCompat("GC inactive nodes tests", (getTestObjectProvider) => {
...
 						...testContainerConfig,
 						loaderProps: { logger: mockLogger },
 					},
 					{ [LoaderHeader.version]: summaryVersion1 },
 				);
+				const defaultDataStoreContainer2 = await requestFluidObject<ITestDataObject>(
+					container2,
+					"/",
+				);
+				defaultDataStoreContainer2._root.set("mode", "write");
```
Comment (tyler-cai-microsoft) [inline packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts:330]: What's the reason behind inactive telemetry not being able to fire for read mode clients?

File: packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts [hunk 2]
```diff
@@ -305,21 +312,21 @@ describeNoCompat("GC data store sweep tests", (getTestObjectProvider) => {
 		itExpects(
 			"Receiving ops for swept datastores fails in client after sweep timeout and summarizing container",
 			[
-				{
-					eventName:
-						"fluid:telemetry:ContainerRuntime:GarbageCollector:SweepReadyObject_Loaded",
```
Comment (tyler-cai-microsoft) [inline packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts:310]: Is this a generic event now and not an error event?

Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts:310]: No, these won't be logged at all. I mentioned this in the PR description:
The other tests were not testing the inactive / sweep ready state but rather tombstone / deleted state. For these, the container does not need to load in "write" mode. So, for these tests removed the expected inactive / sweep ready errors since they won't happen in loaded containers anymore.

Let me know if it makes sense.

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts [hunk 3]
```diff
@@ -890,22 +862,19 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
...
 				{
-					eventName:
-						"fluid:telemetry:ContainerRuntime:GarbageCollector:SweepReadyObject_Loaded",
+					eventName: "fluid:telemetry:ContainerRuntime:GC_Tombstone_DataStore_Requested",
+					category: "generic",
 				},
 				{
-					eventName:
-						"fluid:telemetry:ContainerRuntime:GarbageCollector:GC_Tombstone_Datastore_Revived",
-					category: "generic",
```
Comment (tyler-cai-microsoft) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:906]: We don't have this telemetry anymore?

**Review by tyler-cai-microsoft — APPROVED** [review]
I'm surprised some of the telemetry changed. Looks good otherwise

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts (hunk 3 above)
Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:906]: This one surprised me too. It's very hard to follow what's going on in the test. Could you take a look at this one?

Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:906]: Ok so the first requested telemetry is when we get the first 404. The second one is when we re-request the tombstoned object to store it (that's why it's generic). We should still be getting a GC_tombstone_datastore_revived event. It should be generic. We should get it right before we summarize when we stored the handle of the tombstoned datastore in
`defaultDataObject._root.set("store", tombstonedObject.handle);`

Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:906]: They are logged as expected but the test fails with unexpected GC_Tombstone_Datastore_Revived error. I guess it has something to do with adding generic events in itExpects. Debugging.

Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:906]: Figured it out. I missed adding "GarbageCollector" to the generic event name. But the test was failing with SweepReadyObject_Revived not found so that was confusing.

Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:906]: Nice, feel free to resolve this conversation

**Review by curtisman — APPROVED** [review]
